### ISSUE TITLE
Cannot make react-native-firebase work with react-native-push-notification

### ISSUE THREAD

**Issue by LuongTruong** [invertase/react-native-firebase#3390]
## Issue

For the local notification, react-native-firebase v6 suggest me to use [these libraries](https://invertase.io/oss/react-native-firebase/v6/messaging/quick-start#local-notifications)

[react-native-push-notification](https://github.com/zo0r/react-native-push-notification)... I choose you.

I am able to create a demo project for `react-native-push-notification`, everything works perfectly. However, when I add the below package and run `pod install`, local push notification doesn't working anymore.

- `@react-native-firebase/messaging`

Please take a look guys. Many thanks

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
    "@react-native-community/async-storage": "1.7.1",
    "@react-native-community/cameraroll": "1.3.0",
    "@react-native-community/masked-view": "0.1.5",
    "@react-native-community/netinfo": "5.3.1",
    "@react-native-community/push-notification-ios": "1.1.1",
    "@react-native-firebase/app": "6.4.0-rc4",
    "@react-native-firebase/messaging": "6.4.0-rc4",
    "axios": "0.19.0",
    "moment": "2.24.0",
    "prop-types": "15.7.2",
    "ramda": "0.26.1",
    "react": "16.9.0",
    "react-native": "0.61.5",
    "react-native-background-timer": "2.1.1",
    "react-native-camera": "3.15.1",
    "react-native-device-info": "5.4.0",
    "react-native-fs": "2.16.2",
    "react-native-gesture-handler": "1.5.3",
    "react-native-image-resizer": "1.1.0",
    "react-native-image-zoom-viewer": "2.2.27",
    "react-native-keep-awake": "4.0.0",
    "react-native-orientation-locker": "1.1.8",
    "react-native-pdf": "6.0.1",
    "react-native-push-notification": "^3.1.9",
    "react-native-reanimated": "1.4.0",
    "react-native-safe-area-context": "0.6.2",
    "react-native-screens": "2.0.0-alpha.22",
    "react-native-signature-capture": "0.4.10",
    "react-native-sound": "0.11.0",
    "react-native-splash-screen": "3.2.0",
    "react-native-vector-icons": "6.6.0",
    "react-navigation": "4.0.10",
    "react-navigation-stack": "2.0.8",
    "react-redux": "7.1.3",
    "redux": "4.0.5",
    "redux-promise": "0.6.0",
    "redux-thunk": "2.3.0",
    "rn-fetch-blob": "0.12.0",
    "uuid": "3.3.3"
```

#### `firebase.json` for react-native-firebase v6:

```json
I don't have this file
```
</p>
</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

target 'ProjectName' do
  # Pods for ProjectName
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  # Pods for ProjectName
  pod 'Alamofire'
  pod 'SwiftyJSON'
  pod 'Firebase/Messaging'
  
  use_native_modules!
end
```

#### `AppDelegate.m`:

```objc
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>
#import <UserNotifications/UserNotifications.h>

#if RCT_DEV
#import <React/RCTDevLoadingView.h>
#endif

#import "../../node_modules/react-native-splash-screen/ios/RNSplashScreen.h"

#import "../../node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.h"

// Firebase
#import <Firebase.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  
  if ([FIRApp defaultApp] == nil) {
    [FIRApp configure];
  }
  
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  
  // Fix react-native-device-info warning
  #if RCT_DEV
  [bridge moduleForClass:[RCTDevLoadingView class]];
  #endif
  // ====================================
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                   moduleName:@"ProjectName"
                                            initialProperties:nil];

  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  
  // Splash Screen
  [RNSplashScreen show];
  // =============
  
  // Define UNUserNotificationCenter
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
  
  return YES;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

// Push notification

// You may wonder why there is too little code here. It is exactly the same as the demo project and local notification can be triggered normally
// Called when a notification is delivered to a foreground app.
-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
{
  completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);
}

// Orientation

- (UIInterfaceOrientationMask)application:(UIApplication *)application supportedInterfaceOrientationsForWindow:(UIWindow *)window {
  return [Orientation getOrientation];
}

@end
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 155.25 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 13.3.0 - /usr/local/bin/node
    npm: 6.13.2 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 19, 21, 22, 23, 24, 25, 26, 27, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-22 | Google APIs Intel x86 Atom, android-23 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.6010548
    Xcode: 11.3/11C29 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1

```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
- **`react-native-firebase` version you're using that has this issue:**
  - `v6: 6.4.0-rc4`
- **`Firebase` module(s) you're using that has the issue:**
  - `App`
  - `Messaging`
- **Are you using `TypeScript`?**
  - `Y` & `3.7.4`
  
</p>
</details>

Thank you for the great library
---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by russellwheatley** [comment]
Could you be more specific about what your problem is please? Is there an error for example?

**Comment by LuongTruong** [comment]
Hello @russellwheatley , thank you for considering the problems. I just find out why it is not working. Please take a look at this [link](https://github.com/invertase/react-native-firebase/issues/3367)

I am going to close this issue since we know why it is happening.

**Comment by Salakar** [comment]
Please try 6.4.1-alpha.0 from NPM - this should fix the issues - PR: #3427
